### PR TITLE
Meson: set vpnservicedir path relative to prefix

### PR DESCRIPTION
**Pull request by Soapux**

### CODE REVIEW

**Comment by vhumpa** [comment]
NM CI: Please confirm test execution over this PR. Comment .*ok\W+to\W+test.* to confirm and/or .*add\W+to\W+whitelist.* to add the PR author in whitelist for future.

**Comment by thom311** [comment]
This does not seem right to me

`libnm-core/nm-vpn-plugin-info.c` has

```
#define DEFAULT_DIR_LIB     NMLIBDIR"/VPN"
```
and `./meson.build` has
```
config_extra_h.set_quoted('NMLIBDIR',          nm_pkglibdir)
```

The path that is written into `libnm.pc` certainly must correspond to `DEFAULT_DIR_LIB`. If you want to make that separately configurable, then the path must be also adjusted there.

Can you elaborate what exactly you want to do?

Thank you

**Comment by Soapux** [comment]
Really I was just going by what the autotools pkgconfig file looked like.  The autotools build set the variable as:
vpnservicedir=${prefix}/lib/NetworkManager/VPN

Whereas the current meson just defines an absolute path:

vpnservicedir=/usr/lib/NetworkManager/VPN

**Comment by Soapux** [comment]
What about if your cross-compiling and you build meson with --prefix=/usr --libdir=lib32, wouldn't that variable then be wrong?

**Comment by thom311** [comment]
> What about if your cross-compiling and you build meson with --prefix=/usr --libdir=lib32, wouldn't that 
> variable then be wrong?

--libdir is the location for libraries, however the VPN's `.name` files are not libraries. I think the should always be in `${prefix}/lib/NetworkManager/VPN`.

**Comment by Soapux** [comment]
Ok, I think I got it this time.  Sorry for any noise, I'm still a git novice.

File: libnm/meson.build [hunk 1]
```diff
@@ -184,7 +184,7 @@ pkg.generate(
   requires: 'gio-2.0',
   variables: [
     'exec_prefix=${prefix}',
-    'vpnservicedir=' + join_paths(nm_pkglibdir, 'VPN')
+    'vpnservicedir=' + join_paths('${prefix}', 'lib', nm_name, 'VPN')
   ]
```
[the remark this answers is not in the document]
Comment (lkundrak) [reply libnm/meson.build]: @bengal -- no, they're not equivalent. the first one is relocatable (as in `pkg-config --define-variable=prefix=/whatever`), and that's intentional for the VPN plugins.

I'll apply this, but will attempt to improve the commit message.

**Comment by lkundrak** [comment]
Applied. Thanks!

**Pull request closed without merging**